src/world/region.py

```diff
@@ -33,17 +38,52 @@ def get_neighborhood(self, agent, size=1):
                 neighborhood += self.agents_pos.get((i, j), [])
         return [n for n in neighborhood if n != agent]
 
-    def is_in_map_area(self, x, y):
-        """Check if x, y position is in map"""
-        if self.shape == RegionShape.RECTANGLE.value:
-            if x < 0 or x >= self.width or y < 0 or y >= self.heigth:
-                return False
-            return True
-        else:
-            raise NotImplementedError()
-
-    def generate_pos_from_map(self):
-        if self.shape == RegionShape.RECTANGLE.value:
-            return np.random.randint(self.width), np.random.randint(self.heigth)
-        else:
-            raise NotImplementedError()
+    def is_in_map_area(self, point):
+        """
+        Check if a point is inside a polygon defined by an array of vertices.
+
+        Arguments:
+        point -- a tuple representing the (x, y) coordinates of the test point.
+        vertices -- a list of tuples representing the (x, y) coordinates of the vertices of the polygon.
+
+        Returns:
+        True if the point is inside the polygon, False otherwise.
+        """
+        # Cast a ray from the test point to the right
+        # Count the number of times the ray intersects with the edges of the polygon
+        # If the number of intersections is odd, the point is inside the polygon
+        # If the number of intersections is even, the point is outside the polygon
+
+        num_intersections = 0
+        for i in range(len(self.vertices)):
+            j = (i + 1) % len(self.vertices)
+            if ((self.vertices[i][1] > point[1]) != (self.vertices[j][1] > point[1])) and (
+                    point[0] < (self.vertices[j][0] - self.vertices[i][0]) * (point[1] - self.vertices[i][1]) / (
+                    self.vertices[j][1] - self.vertices[i][1]) + self.vertices[i][0]):
+                num_intersections += 1
+
+        return num_intersections % 2 == 1
```

Comment:
Why didn't you use `shapely.geometry.Polygon` and method `contains` as in `_generate_agent_pos_inside_region`?